pkg/odo/cli/component/delete.go

```diff
@@ -59,7 +60,12 @@ func (do *DeleteOptions) Run() (err error) {
 	glog.V(4).Infof("component delete called")
 	glog.V(4).Infof("args: %#v", do)
 
-	if do.componentForceDeleteFlag || ui.Proceed(fmt.Sprintf("Are you sure you want to delete %v from %v?", do.componentName, do.Application)) {
+	if do.outputFormat == "json" {
```

Comment:
what if someone provides invalid -o format? see https://github.com/redhat-developer/odo/pull/1404/files#r261588399

Reply:
rebased and the resolved this issue